src/Umbraco.Core/Migrations/Upgrade/V_8_0_0/VariantsMigration.cs

```diff
@@ -90,6 +91,15 @@ private void MigratePropertyData()
             Rename.Table(PreTables.PropertyData).To(Constants.DatabaseSchema.Tables.PropertyData).Do();
         }
 
+        private void CreatePropertyDataIndexes()
+        {
+            // Creates a temporary index on umbracoPropertyData to speed up other migrations which update property values.
+            // It will be removed in CreateKeysAndIndexes before the normal indexes for the table are created
+            var tableDefinition = Persistence.DatabaseModelDefinitions.DefinitionFactory.GetTableDefinition(typeof(PropertyDataDto), SqlSyntax);
+            Execute.Sql(SqlSyntax.FormatPrimaryKey(tableDefinition)).Do();
+            Execute.Sql($"CREATE UNIQUE NONCLUSTERED INDEX IX_umbracoPropertyData_Temp ON {PreTables.PropertyData} (versionId,propertyTypeId,languageId,segment)").Do();
```

Comment:
Makes sense. I'm fairly positive that this is the structure that we actually want the index to exist on this table anyways instead of having 4x individual indexes which i think not actually helping us at all and is probably causing slower performance than faster perf. Something we should look into soon. Have created a separate issue for this https://github.com/umbraco/Umbraco-CMS/issues/7570

Reply:
I'll need to fix this up - at this stage this will throw an exception because we've already changed the table name in `MigratePropertyData`. I'm also unsure why the format PK line is here?